### PR TITLE
doc: lint and modernize code examples in guides

### PR DESCRIPTION
**Pull request by vsemozhetbyt**
1. Lint code examples in guides according to the main repository rules (also add some `eslint-plugin-markdown` comments to ease possible future linting).

2. Modernize some fragments (arrow functions in callbacks, destructuring, template literals, object shorthand, `Buffer.alloc()` etc).

Some trailing spaces were deleted automatically.

### CODE REVIEW

File: locale/en/docs/guides/domain-postmortem.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ the active domain to hijack the error:
 const domain = require('domain');
 const net = require('net');
 const d = domain.create();
-d.on('error', (err) => console.error(err.message));
+d.on('error', (err) => { console.error(err.message); });
```
Comment (lpinca) [inline locale/en/docs/guides/domain-postmortem.md]: I think the original is cleaner.

File: locale/en/docs/guides/domain-postmortem.md [hunk 2]
```diff
@@ -127,16 +127,16 @@ d1.on('error', (er) => { /* handle error */ });
...
-  d2.on('error', (er) => console.error(er.message, domain._stack));
+  d2.on('error', (er) => { console.error(er.message, domain._stack); });
   d2.run(() => {
     setTimeout(() => {
       setTimeout(() => {
         throw new Error('outer');
-      });
-      throw new Error('inner')
-    });
+      }, 1);
```
Comment (lpinca) [inline locale/en/docs/guides/domain-postmortem.md]: This is what we use in core but I think it creates more confusion than it tries to solve here

File: locale/en/docs/guides/backpressuring-in-streams.md [hunk 3]
```diff
@@ -42,11 +42,10 @@ const readline = require('readline');
 // process.stdin and process.stdout are both instances of Streams
 const rl = readline.createInterface({
   input: process.stdin,
-  output: process.stdout
+  output: process.stdout,
```
Comment (lpinca) [inline locale/en/docs/guides/backpressuring-in-streams.md]: Why the lax comma?

File: locale/en/docs/guides/domain-postmortem.md [hunk 4]
```diff
@@ -360,8 +360,8 @@ const server = net.createServer((c) => {
   // Mock class that does some useless async data transformation
   // for demonstration purposes.
   const ds = new DataStream(dataTransformed);
-  c.on('data', (chunk) => ds.data(chunk));
-}).listen(8080, () => console.log(`listening on 8080`));
+  c.on('data', (chunk) => { ds.data(chunk); });
```
Comment (lpinca) [inline locale/en/docs/guides/domain-postmortem.md]: Why this change?

File: locale/en/docs/guides/domain-postmortem.md (hunk 1 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: It seems to me that omission of curly braces infers implicit return. Let me know if I should revert.

File: locale/en/docs/guides/domain-postmortem.md (hunk 2 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: Let me know if I should revert.

File: locale/en/docs/guides/domain-postmortem.md (hunk 4 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: It seems to me that omission of curly braces infers implicit return. Let me know if I should revert.

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 3 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/backpressuring-in-streams.md]: It is common in API docs and the rest code base (mostly tests?) if I recall correctly.  Let me know if I should revert.

File: locale/en/docs/guides/domain-postmortem.md (hunk 1 above)
Comment (lpinca) [reply locale/en/docs/guides/domain-postmortem.md]: The two are equivalent as `console.error()` returns `undefined` no?

File: locale/en/docs/guides/domain-postmortem.md (hunk 2 above)
Comment (lpinca) [reply locale/en/docs/guides/domain-postmortem.md]: This is only my opinion so feel free to do that.

File: locale/en/docs/guides/domain-postmortem.md (hunk 4 above)
Comment (lpinca) [reply locale/en/docs/guides/domain-postmortem.md]: The returned value is ignored so it doesn't matter what the function returns.

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 3 above)
Comment (lpinca) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Feel free to do that, if you think we should use lax commas keep it and maybe wait for other feedbacks?

File: locale/en/docs/guides/domain-postmortem.md (hunk 1 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: Yes, they are, but implicit return infers it is used somehow (as in `.map()` callbacks etc). Maybe I am wrong here. Will be happy to revert if it is more confusing.

File: locale/en/docs/guides/domain-postmortem.md (hunk 2 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: It is in some other changes too, so I am not sure. If we lint this folder when new guides appear, then these fixes will ease the process. However, if they are confusing, the reading issue is maybe more important than the meta-processes issue.

File: locale/en/docs/guides/domain-postmortem.md (hunk 4 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: Yes, but personally I should every time to check if it is ignored without curlies. However, it may be my personal bias, so I will be happy to revert if it is more common.

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 3 above)
Comment (fhemberger) [reply locale/en/docs/guides/backpressuring-in-streams.md]: -1 on lax-commas

Comment (hiroppy) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Agree with @fhemberger.

File: locale/en/docs/guides/domain-postmortem.md (hunk 4 above)
Comment (lpinca) [reply locale/en/docs/guides/domain-postmortem.md]: AFAIK the return value is always ignored when the function is a listener of an event due to how the `EventEmitter` works.

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 3 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Reverted.

File: locale/en/docs/guides/domain-postmortem.md (hunk 2 above)
Comment (lpinca) [reply locale/en/docs/guides/domain-postmortem.md]: A reader not familiar with how timers work may think that there is a difference between `setTimeout(onTimeout, 0)` and  `setTimeout(onTimeout, 1)` and may wonder why `1` is used here instead of `0`. This is basically the reason for my comment.

File: locale/en/docs/guides/domain-postmortem.md [hunk 5]
```diff
@@ -383,15 +383,15 @@ DataStream.prototype.data = function data(chunk) {
   // This code is self contained, but pretend it's a complex
   // operation that crosses at least one other module. So
   // passing along "this", etc., is not easy.
-  this.domain.run(function() {
+  this.domain.run(() => {
```
Comment (fhemberger) [inline locale/en/docs/guides/domain-postmortem.md]: Not necessarily part of this PR, but `domain` is deprecated. We should probably remove this.

File: locale/en/docs/guides/event-loop-timers-and-nexttick.md [hunk 6]
```diff
@@ -338,18 +334,18 @@ This philosophy can lead to some potentially problematic situations.
 Take this snippet for example:
 
 ```js
+let bar;
```
Comment (fhemberger) [inline locale/en/docs/guides/event-loop-timers-and-nexttick.md:337]: Why are you initializing the variable up here instead of assigning it directly as before?

**Review by hiroppy — APPROVED** [review]
Rubber-stamp LGTM 👍  Thanks @vsemozhetbyt !

File: locale/en/docs/guides/event-loop-timers-and-nexttick.md [hunk 7]
```diff
@@ -368,15 +364,17 @@ useful for the user to be alerted to an error before the event loop is
 allowed to continue. Here is the previous example using `process.nextTick()`:
 
 ```js
-function someAsyncApiCall (callback) {
+let bar;
```
Comment (fhemberger) [inline locale/en/docs/guides/event-loop-timers-and-nexttick.md:367]: Why are you initializing the variable up here instead of assigning it directly as before?

**Comment by fhemberger** [comment]
LGTM.

Are these changes also required in other locales?

File: locale/en/docs/guides/domain-postmortem.md [hunk 8]
```diff
@@ -127,13 +127,13 @@ d1.on('error', (er) => { /* handle error */ });
 d1.run(() => setTimeout(() => {
   const d2 = domain.create();
   d2.bar = 43;
-  d2.on('error', (er) => console.error(er.message, domain._stack));
+  d2.on('error', (er) => { console.error(er.message, domain._stack); });
```
Comment (lpinca) [inline locale/en/docs/guides/domain-postmortem.md]: Sorry to pester, can you please also revert this for consistency?

**Review by lpinca — APPROVED** [review]

File: locale/en/docs/guides/event-loop-timers-and-nexttick.md (hunk 6 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/event-loop-timers-and-nexttick.md:337]: `var` declaration is hoisted, with `let` declaration in the same place we have `ReferenceError: bar is not defined`

File: locale/en/docs/guides/event-loop-timers-and-nexttick.md (hunk 7 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/event-loop-timers-and-nexttick.md:367]: The same as above.

File: locale/en/docs/guides/domain-postmortem.md (hunk 8 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/domain-postmortem.md]: Reverted in the same commit.

**Comment by vsemozhetbyt** [comment]
@fhemberger It seems we have guides only in `ko` and `uk` locales. Let me know if I should address them.

File: locale/en/docs/guides/event-loop-timers-and-nexttick.md (hunk 6 above)
Comment (fhemberger) [reply locale/en/docs/guides/event-loop-timers-and-nexttick.md:337]: Ah, ok. Didn't notice that. 👍

**Comment by fhemberger** [comment]
@vsemozhetbyt That would be **super** awesome, if you could copy your code changes to those two locales as well. 💯

**Comment by vsemozhetbyt** [comment]
@fhemberger Should it be done in a separate PR?

**Comment by fhemberger** [comment]
@vsemozhetbyt I don't have any preference on that. Whatever suites you best.

**Comment by vsemozhetbyt** [comment]
@fhemberger I shall do another PR a bit later for the other locales.

**Comment by fhemberger** [comment]
Awesome, thank you for this! 👍
